components/motor/gpio/controlled.go

```diff
@@ -259,15 +259,43 @@ func (cm *controlledMotor) GoTo(ctx context.Context, rpm, targetPosition float64
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
 func (cm *controlledMotor) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(cm.Name().ShortName())
+	cm.opMgr.CancelRunning(ctx)
+	ctx, done := cm.opMgr.New(ctx)
+	defer done()
+
+	warning, err := checkSpeed(rpm, cm.real.maxRPM)
+	if warning != "" {
+		cm.logger.CWarnf(ctx, warning)
+	}
+	if err != nil {
+		return err
+	}
+
+	if cm.loop == nil {
+		// create new control loop
+		if err := cm.startControlLoop(); err != nil {
+			return err
+		}
+	}
+
+	cm.loop.Resume()
+	// set control loop values
+	velVal := math.Abs(rpm * cm.ticksPerRotation / 60)
```

Comment:
We're now calling `motor.ticksperRotation` in several places out of lock. This could introduce races with reconfiguration, or GoFor with SetRPM.

Reply:
I'll go through and make sure it's being locked everywhere